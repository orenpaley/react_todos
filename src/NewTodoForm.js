import React, { useState } from "react";
import { v4 as uuid } from "uuid";
// creates an input controlled by react to add text and create new todo
const NewTodoForm = ({ addTodo }) => {
  const INITIAL_STATE = {
    msg: "write a new task",
    isComplete: false,
  };
  const [formData, setFormData] = useState(INITIAL_STATE);
  const handleChange = (e) => {
    const { name, value } = e.target;
    console.log("formdataB4", formData);
    console.log("name, value", name, value);

    setFormData(() => ({
      // maintains state of other data that is not being changed
      ...formData,
      // changes state of the value of name -> msg
      [name]: value,
    }));
    console.log("formdataAFter", formData);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({ ...formData, key: uuid() });
    setFormData(INITIAL_STATE);
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="msg">Task</label>
      <input
        id="msg"
        type="text"
        name="msg"
        placeholder="write a new task"
        value={formData.msg}
        onChange={handleChange}
      />
      <button>+</button>
    </form>
  );
};

export default NewTodoForm;
