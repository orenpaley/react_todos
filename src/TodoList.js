import { useState } from "react";
import { v4 as uuid } from "uuid";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
  let INITIAL_STATE = [
    { msg: "i am a todo", isComplete: true, id: uuid() },
    { msg: "i am a 2nd todo", isComplete: false, id: uuid() },
    { msg: "i am a 3rd todo", isComplete: false, id: uuid() },
  ];

  const [Todos, SetTodos] = useState(INITIAL_STATE);
  const addTodo = (newTodo) => {
    SetTodos((todos) => [...todos, { ...newTodo, id: uuid() }]);
  };
  const removeTodo = (todoToRemove) => {
    SetTodos((todos) => todos.filter((t) => t.id !== todoToRemove.id));
  };
  const completeTodo = (todoToComplete) => {
    let newTodos = [...Todos];
    let changed = newTodos.findIndex((todo) => todo.id === todoToComplete.id);
    console.log("changed b4", newTodos[changed]);

    newTodos[changed].isComplete === true
      ? (newTodos[changed].isComplete = false)
      : (newTodos[changed].isComplete = true);
    console.log("changed after", newTodos[changed]);
    SetTodos(newTodos);
    // let newTodos = Todos.filter((t) => t.id !== todoToComplete.id);
    // SetTodos([...newTodos, Todos[changed]]);
  };
  return (
    <div>
      <h1>Todo List</h1>
      <div>
        <NewTodoForm addTodo={addTodo} />
      </div>
      {/* <NewTodoForm addTodo={addTodo} /> */}
      <div className="TodoList">
        {Todos.map(({ msg, isComplete, id }) => (
          <Todo
            key={id}
            id={id}
            msg={msg}
            isComplete={isComplete}
            remove={removeTodo}
            complete={completeTodo}
          />
        ))}
      </div>
    </div>
  );
};

export default TodoList;
