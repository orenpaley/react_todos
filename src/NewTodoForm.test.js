import NewTodoForm from "./NewTodoForm";
import { addTodo } from "./TodoList";

import { render } from "@testing-library/react";

it("renders without crashing", () => {
  render(<NewTodoForm addTodo={addTodo} />);
});
it("matches snapshot", function () {
  const { asFragment } = render(<NewTodoForm addTodo={addTodo} />);
  expect(asFragment()).toMatchSnapshot();
});
