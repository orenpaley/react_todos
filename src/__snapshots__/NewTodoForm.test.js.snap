// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="msg"
    >
      Task
    </label>
    <input
      id="msg"
      name="msg"
      placeholder="write a new task"
      type="text"
      value="write a new task"
    />
    <button>
      +
    </button>
  </form>
</DocumentFragment>
`;
